src/hooks/useGetUserPlayStreak.ts

```diff
@@ -1,26 +1,36 @@
 import { getGetUserPlayStreakQueryKey } from '@/helpers/getQueryKeys'
 import { UserPlayStreak } from '@hyperplay/utils'
-import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { queryOptions, useQuery, useQueryClient } from '@tanstack/react-query'
 
-export function useGetUserPlayStreak(
+export function getUserPlaystreakQueryOptions(
   questId: number | null,
   getUserPlayStreak: (questId: number) => Promise<UserPlayStreak>
 ) {
-  const queryClient = useQueryClient()
   const queryKey = getGetUserPlayStreakQueryKey(questId)
-  const query = useQuery({
+  return queryOptions({
     queryKey,
     queryFn: async () => {
       if (questId === null) {
         return null
       }
       const response = await getUserPlayStreak(questId)
       if (!response) return null
-      return response
+      return { questId: questId, userPlayStreak: response }
     },
     refetchOnWindowFocus: false,
     enabled: questId !== null
   })
+}
+
+export function useGetUserPlayStreak(
+  questId: number | null,
+  getUserPlayStreak: (questId: number) => Promise<UserPlayStreak>
+) {
+  const queryClient = useQueryClient()
+  const queryKey = getGetUserPlayStreakQueryKey(questId)
```

Comment:
Why are we returning the whole query under data? Also, I think we can get the queryKey from the options directly.

```ts
const queryOptions = getUserPlaystreakQueryOptions(questId, getUserPlayStreak);
const query = useQuery(queryOptions)

return {
    ...query,
    isLoading: query.isLoading || query.isFetching,
    invalidateQuery: async () => queryClient.invalidateQueries({ queryKey: queryOptions.queryKey })
  }
```

Reply:
This is the pattern we use elsewhere in the client. I think we should keep the query props separated from the top level of this object so we don't overwrite them like we would be doing here with `isLoading`. It is a relatively minor stylistic difference.

Just updated to get the query key from the options. Good call.